tool/e2e/test/brick/brick_test.dart

```diff
@@ -0,0 +1,79 @@
+@Tags(['e2e'])
+library e2e;
+
+import 'dart:io';
+
+import 'package:mason/mason.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+/// Exit code indicating a command completed successfully.
+///
+/// [Source](https://www.freebsd.org/cgi/man.cgi?query=sysexits).
+const _sucessfulExitCode = 0;
+
+void main() {
+  group('very_good_core brick', () {
+    test(
+      'creates a tested application',
+      timeout: const Timeout(Duration(minutes: 2)),
+      () async {
+        final rootPath = Directory.current.parent.parent.path;
+        final brickPath = path.join(rootPath, 'brick');
+        final brick = Brick.path(brickPath);
+        final masonGenerator = await MasonGenerator.fromBrick(brick);
+        final tempDirectory = Directory.systemTemp.createTempSync();
+        final directoryGeneratorTarget = DirectoryGeneratorTarget(
+          tempDirectory,
+        );
+        final vars = <String, dynamic>{
+          'project_name': 'test_app',
+          'org_name': 'very_good_ventures',
+          'application_id': 'verygood.ventures.test',
+          'description': 'very_good_core test',
+        };
+        await masonGenerator.generate(directoryGeneratorTarget, vars: vars);
+
+        final applicationPath = path.join(
+          directoryGeneratorTarget.dir.path,
+          vars['project_name'] as String,
+        );
+        final flutterPubGetResult = await Process.run(
+          'flutter',
+          ['pub', 'get'],
+          workingDirectory: applicationPath,
+          runInShell: true,
+        );
+        expect(
+          flutterPubGetResult.exitCode,
+          equals(_sucessfulExitCode),
+          reason: '`flutter pub get` failed',
+        );
+        expect(
+          flutterPubGetResult.stderr,
+          isEmpty,
+          reason: '`flutter pub get` failed',
+        );
+
+        final flutterTest = await Process.run(
+          'flutter',
+          ['test'],
+          workingDirectory: applicationPath,
+          runInShell: true,
+        );
+        expect(
+          flutterTest.exitCode,
+          equals(_sucessfulExitCode),
+          reason: '`flutter test` failed',
+        );
+        expect(
+          flutterTest.stderr,
+          isEmpty,
+          reason: '`flutter test` failed',
+        );
+
+        tempDirectory.deleteSync(recursive: true);
```

Comment:
Should we also test this against a fixture? Or is that already handled in a different test? 

/CC @renancaraujo

Reply:
I think what er have already is pretty much safe. Systemp exists pretty much for this kind of use case

Reply:
I agree with both of you.

I think testing against a fixture might be helpful, but I think that test can be added on another PR.

Reply:
Yes, let's do that in a follow-up PR for #220